### ISSUE TITLE
Cyclic dependency between Flags and Logging (RequestContextExportingAppender)

### ISSUE THREAD

**Issue by hugobellomusto** [line/armeria#4287]
I was trying the last armeria release ([1.16.0](https://armeria.dev/release-notes/1.16.0/)) and "something strange" happens in a simple hello world app when using the `RequestContextExportingAppender` .

To simplify, if you have a main with only this:
```scala
@main def main()=
  println(Flags.requestContextStorageProvider())
```
(the same happens with useful code like `Server.builder.http(8080)…blabla` )

And logback is configured with the `RequestContextExportingAppender` as it is described in the [docs](https://armeria.dev/docs/advanced-logging/).

**Apps crashes at start-up** with:
```
[error] Exception in thread "main" java.lang.ExceptionInInitializerError
[error] 	at com.linecorp.armeria.common.RequestContext.currentOrNull(RequestContext.java:85)
[error] 	at com.linecorp.armeria.common.logging.RequestContextExporter.export(RequestContextExporter.java:204)
[error] 	at com.linecorp.armeria.common.logback.RequestContextExportingAppender.append(RequestContextExportingAppender.java:189)
[error] 	at com.linecorp.armeria.common.logback.RequestContextExportingAppender.append(RequestContextExportingAppender.java:57)
[error] 	at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:84)
[error] 	at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:51)
[error] 	at ch.qos.logback.classic.Logger.appendLoopOnAppenders(Logger.java:270)
[error] 	at ch.qos.logback.classic.Logger.callAppenders(Logger.java:257)
[error] 	at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:421)
[error] 	at ch.qos.logback.classic.Logger.filterAndLog_0_Or3Plus(Logger.java:383)
[error] 	at ch.qos.logback.classic.Logger.info(Logger.java:591)
[error] 	at com.linecorp.armeria.common.Flags.getValue(Flags.java:1392)
[error] 	at com.linecorp.armeria.common.Flags.getValue(Flags.java:1377)
[error] 	at com.linecorp.armeria.common.Flags.<clinit>(Flags.java:108)
[error] 	at daap.apiregistry.application.Main$package$.main(Main.scala:8)
[error] 	at daap.apiregistry.application.main.main(Main.scala:7)
[error] Caused by: java.lang.IllegalStateException: Failed to create context storage. provider: null
[error] 	at com.linecorp.armeria.internal.common.RequestContextUtil.<clinit>(RequestContextUtil.java:69)
[error] 	... 16 more
[error] Caused by: java.lang.NullPointerException
[error] 	at com.linecorp.armeria.internal.common.RequestContextUtil.<clinit>(RequestContextUtil.java:67)
[error] 	... 16 more
```
Debugging, I found what seems to be a **cyclic dependency** between Flags and Logging (see attached image). It is something like this (summarized _call stack_):
```
Flags.requestContextStorageProvider()...   // This method returns null, because REQUEST_CONTEXT_STORAGE_PROVIDER static var has not been initialized yet in the class Flags (because it is initializing)
...
RequestContextExportingAppender... // It need some info from class Flags that is initializing
...
Logger.info // Flags class wanted to log useful info
Flags.<init> // java class loader initializing class Flags and running static code
main()
```

![image](https://user-images.githubusercontent.com/2941566/172388139-ba79c8cb-7575-42a1-92b1-acdf6be26804.png)

**Comment by hugobellomusto** [comment]
The workaround is to turn off Flags logging in`logback.xml` with:
 
```xml
 <logger name="com.linecorp.armeria.common.Flags" level="OFF" />
```

**Comment by ikhoon** [comment]
Fixed by #4285
